### PR TITLE
recognise e2e format when intent restated as /intent

### LINKED ISSUE #4520: e2e evaluation raises ZeroDivisionError
**Rasa version**: 1.3.6

**Python version**: 3.6

**Operating system** (windows, osx, ...): osx

**Issue**:

e2e story evaluation always raises `ZeroDivisionError`

**Error (including full traceback)**:
```
Processed Story Blocks: 0it [00:00, ?it/s]
2019-09-25 15:30:06 WARNING  rasa.core.training.generator  - There is no starting story block in the training data. All your story blocks start with some checkpoint. There should be at least one story block that starts without any checkpoint.
2019-09-25 15:30:06 INFO     rasa.core.test  - Evaluating 0 stories
Progress:
0it [00:00, ?it/s]
2019-09-25 15:30:06 INFO     rasa.core.test  - Finished collecting predictions.
2019-09-25 15:30:06 WARNING  rasa.nlu.test  - No labels to evaluate. Skip evaluation.
Traceback (most recent call last):
  File "/Users/ric/.envs/platform/bin/rasa", line 11, in <module>
    load_entry_point('rasa', 'console_scripts', 'rasa')()
  File "/Users/ric/dev/rasa/rasa/__main__.py", line 76, in main
    cmdline_arguments.func(cmdline_arguments)
  File "/Users/ric/dev/rasa/rasa/cli/test.py", line 88, in test_core
    kwargs=vars(args),
  File "/Users/ric/dev/rasa/rasa/test.py", line 108, in test_core
    rasa.core.test(stories, _agent, out_directory=output, **kwargs)
  File "uvloop/loop.pyx", line 1448, in uvloop.loop.Loop.run_until_complete
  File "/Users/ric/dev/rasa/rasa/core/test.py", line 483, in test
    completed_trackers, agent, fail_on_prediction_errors, e2e
  File "/Users/ric/dev/rasa/rasa/core/test.py", line 430, in collect_story_predictions
    in_training_data_fraction = _in_training_data_fraction(action_list)
  File "/Users/ric/dev/rasa/rasa/core/test.py", line 383, in _in_training_data_fraction
    return len(in_training_data) / len(action_list)
ZeroDivisionError: division by zero
```

**Command or request that led to error**:
```
$ rasa test core --stories e2e.md --e2e
```

### PR DESCRIPTION
**Pull request by melindaloubser1**
**Proposed changes**:
- modify retrieval action story regex to exclude matching e2e stories
- closes #4520 
@wochinge 

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/nlu/training_data/loading.py [hunk 1]
```diff
@@ -51,7 +51,7 @@
 # ##
 # * intent/response_key
 #   - response_text
-_nlg_markdown_marker_regex = re.compile(r"##\s*.*\n\*.*\/.*\n\s*\t*\-.*")
+_nlg_markdown_marker_regex = re.compile(r"##\s*.*\n\*.*(?<!: )\/.*\n\s*\t*\-.*")
```
Comment (wochinge) [inline rasa/nlu/training_data/loading.py]: I'm not a big regex expert, but I think this one would be a bit simpler:
```suggestion
_nlg_markdown_marker_regex = re.compile(r"##\s*.*\n\*[^:]*\/.*\n\s*\t*\-.*")
```

File: data/test_evaluations/end_to_end_story.md [hunk 2]
```diff
@@ -24,4 +24,8 @@
 * deny: no
  - utter_ack_dosearch
  - action_search_restaurants
- - action_suggest
\ No newline at end of file
+ - action_suggest
+
+## story_with_intent_restated
+* greet: /greet
```
Comment (wochinge) [inline data/test_evaluations/end_to_end_story.md]: great work in adding a test for this as well 🚀

File: rasa/nlu/training_data/loading.py (hunk 1 above)
Comment (melindaloubser1) [reply rasa/nlu/training_data/loading.py]: Yeah that makes sense

**Comment by wochinge** [comment]
@melindaloubser1 Thanks for your well-tested contribution. We gonna release it today with the next patch release 🎉

**Comment by wochinge** [comment]
@erohmensing Please move the changelog to the correct position when you do the release 🙏

**Pull request merged**